x-pack/test/functional/page_objects/api_keys_page.ts

```diff
@@ -26,5 +27,70 @@ export function ApiKeysPageProvider({ getService }: FtrProviderContext) {
     async apiKeysPermissionDeniedMessage() {
       return await testSubjects.getVisibleText('apiKeysPermissionDeniedMessage');
     },
+
+    async clickOnPromptCreateApiKey() {
+      return await testSubjects.click('apiKeysCreatePromptButton');
+    },
+
+    async clickOnTableCreateApiKey() {
+      return await testSubjects.click('apiKeysCreateTableButton');
+    },
+
+    async setApiKeyName(apiKeyName: string) {
+      return await testSubjects.setValue('apiKeyNameInput', apiKeyName);
+    },
+
+    async setApiKeyCustomExpiration(expirationTime: string) {
+      return await testSubjects.setValue('apiKeyCustomExpirationInput', expirationTime);
+    },
+
+    async submitOnCreateApiKey() {
+      return await testSubjects.click('formFlyoutSubmitButton');
+    },
+
+    async isApiKeyModalExists() {
+      return await find.existsByCssSelector('[role="dialog"]');
+    },
+
+    async getNewApiKeyCreation() {
+      const euiCallOutHeader = await find.byCssSelector('.euiCallOutHeader__title');
+      return euiCallOutHeader.getVisibleText();
+    },
+
+    async toggleCustomExpiration() {
+      return await testSubjects.click('apiKeyCustomExpirationSwitch');
+    },
+
+    async getErrorCallOutText() {
+      const alertElem = await find.byCssSelector('[role="dialog"] [role="alert"] .euiText');
+      return await alertElem.getVisibleText();
+    },
+
+    async getApiKeysFirstPromptTitle() {
+      const titlePromptElem = await find.byCssSelector('.euiEmptyPrompt .euiTitle');
+      return await titlePromptElem.getVisibleText();
+    },
+
+    async deleteAllApiKeyOneByOne() {
+      const hasApiKeysToDelete = await testSubjects.exists('apiKeysTableDeleteAction');
+      if (hasApiKeysToDelete) {
+        const apiKeysToDelete = await testSubjects.findAll('apiKeysTableDeleteAction');
+        for (const element of apiKeysToDelete) {
+          await element.click();
+          await testSubjects.click('confirmModalConfirmButton');
+        }
+      }
+    },
+
+    async bulkDeleteApiKeys() {
+      const hasApiKeysToDelete = await find.existsByCssSelector(
```

Comment:
Note for Reviewers -> I tried to use `testSubjects.exists` here but for some reason it was never happy with it. So I went the old fashion way and it is working 😕

Reply:
@thomheymann, thank you for keeping me on my toes and now it is much better